paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/FlinkTableSink.java

```diff
@@ -201,7 +196,10 @@ private void checkDeletable() {
                             "merge engine '%s' can not support delete, currently only %s can support delete.",
                             options.get(MERGE_ENGINE), MergeEngine.DEDUPLICATE));
         } else if (table instanceof ChangelogValueCountFileStoreTable) {
-            return;
+            throw new UnsupportedOperationException(
```

Comment:
PR https://github.com/apache/incubator-paimon/pull/1216 made this changed. I'm not sure why roll it back.